server/controllers/incidents.js

```diff
@@ -3,53 +3,46 @@ const User = require("../models").Users;
 const Location = require("../models").Locations;
 const Category = require("../models").Categories;
 const Status = require("../models").Statuses;
+const LocationService = require("../services/locations");
 
 module.exports = {
   //create an incident
   create(req, res) {
     let locationPromise;
-    name = req.body.name;
-    centre = req.body.centre;
-    country = req.body.country;
-    return Location
-      .findOne({ where: { name, centre, country } })
-      .then((location) => {
-        (!location) ?
-          locationPromise =
-          Location
-            .create({
-              name: req.body.name,
-              centre: req.body.centre,
-              country: req.body.country
-            })
-            .then(location => {
-              return Promise.resolve(location.dataValues.id);
-            })
-            .catch(error => {
-              res.status(400).send(error);
-            })
-          :
-          locationPromise = Promise.resolve(location.dataValues.id);
-        locationPromise.then(locationId => {
-          Incident
-            .create({
-              description: req.body.description,
-              subject: req.body.subject,
-              dateOccurred: req.body.dateOccurred,
-              userId: req.body.userId,
-              statusId: req.body.statusId || 1,
-              locationId,
-              categoryId: req.body.categoryId,
-              levelId: req.body.levelId,
-              witnesses: req.body.witnesses
-            })
-            .then(incident => {
-              res.status(201).send({ data: incident, status: "success" });
-            }).catch(error => {
-              res.status(400).send(error);
-            });
+    name = req.body.name,
+    centre = req.body.centre,
+    country = req.body.country
+    return Location.findOne({ where: { name, centre, country } })
+    .then(location => {
+      (!location) ?
+      locationPromise =
+      LocationService.create(name, centre, country).then(location => {
+        return Promise.resolve(location.dataValues.id);
+      }).catch(error => {
+        res.status(400).send(error);
+      })
+      :
+      locationPromise = Promise.resolve(location.dataValues.id);
+      locationPromise.then(locationId => {
+        Incident
+        .create({
+          description: req.body.description,
+          subject: req.body.subject,
+          dateOccurred: req.body.dateOccurred,
+          userId: req.body.userId,
+          statusId: req.body.statusId || 1,
+          locationId,
+          categoryId: req.body.categoryId,
```

Comment:
this too

Reply:
the user can just enter the subject. We need to minimize necessary input for better User Experience.

Reply:
@bmuthoga can these changes also be reflected in the bot?

Reply:
Updated. @Mnickii